tsdb/chunkenc/histogram_meta.go

```diff
@@ -140,6 +175,43 @@ func readZeroThreshold(br *bstreamReader) (float64, error) {
 	}
 }
 
+// putCustomBound writes the custom bound to the bstream. It stores whole number
+// values from 0 to 254 (inclusive) in just one byte, but needs 9 bytes for other
+// values like negative numbers, whole numbers greater than or equal to 255, or
+// floats. In detail:
+//   - If the bound is >= 0 and <= 254, store it as a byte.
+//   - Otherwise, store 255 as a single byte, followed by the 8 bytes of
+//     the bound as a float64, i.e. taking 9 bytes in total.
+//
+// This assumes that negative bounds, bounds >= 255, and non-whole number bounds
+// are less common.
+func putCustomBound(b *bstream, f float64) {
+	if f < 0 || f > 254 || f != math.Trunc(f) {
```

Comment:
Have you managed to do some tests with various encoding and some real life examples?

Reply:
Just a random thought from the sideline: I would expect fractional number to be quite common (e.g. latency histogram in seconds, most buckets will be sub-second and therefore something like 0.5, 0.01, etc.).

What we can say is that humans tend to specify bounds that don't have a lot of digits in decimal representations. So what about multiplying the bounds by 100 and then use the trick you are doing here? Then everything between 0.01 and 2.54 (in steps of 0.01) would be encoded as one byte.